### PR TITLE
Update to changes in `BatchNormalize`

### PR DESCRIPTION
**Pull request by daxpedda**
This updates to changes made in `BatchNormalize` in https://github.com/RustCrypto/traits/pull/1896. I also added a test for `NonIdentity::batch_normalize` for `p256`.

I looked into moving this test into `test_projective_arithmetic`, but I don't see a good way to currently do this because of how we can't get that array type for `prop_compose`. Alternatively we could require the corresponding types to implement [`Strategy`](https://docs.rs/proptest/latest/proptest/strategy/trait.Strategy.html). `proptest` also just updated to `rand` v0.9 so I think this would be a good idea. WDYT?

Depends on https://github.com/RustCrypto/traits/pull/1896.

### CODE REVIEW

**Comment by daxpedda** [comment]
Closing in favor of https://github.com/RustCrypto/elliptic-curves/pull/1257.
